src/app/tests/TestExtensionFieldSets.cpp

```diff
@@ -68,118 +76,118 @@ void TestInsertExtensionFieldSet(nlTestSuite * aSuite, void * aContext)
     // Confirm EFS empty
-    NL_TEST_ASSERT(aSuite, tempEFS.mUsedBytes == 0);
-    NL_TEST_ASSERT(aSuite, !memcmp(empty_buffer, tempEFS.mBytesBuffer, sizeof(empty_buffer)));
+    EXPECT_EQ(tempEFS.mUsedBytes, 0);
+    EXPECT_EQ(memcmp(empty_buffer, tempEFS.mBytesBuffer, sizeof(empty_buffer)), 0);
 
     // Test creation of EFS from truncating an Array
     tempEFS = scenes::ExtensionFieldSet(kColorControlClusterId, double_size_buffer, sizeof(tempEFS.mBytesBuffer));
-    NL_TEST_ASSERT(aSuite, tempEFS.mID == kColorControlClusterId);
+    EXPECT_EQ(tempEFS.mID, kColorControlClusterId);
     // Confirm EFS was written
-    NL_TEST_ASSERT(aSuite, tempEFS.mUsedBytes == static_cast<uint8_t>(sizeof(tempEFS.mBytesBuffer)));
-    NL_TEST_ASSERT(aSuite, !memcmp(double_size_buffer, tempEFS.mBytesBuffer, sizeof(tempEFS.mBytesBuffer)));
+    EXPECT_EQ(tempEFS.mUsedBytes, static_cast<uint8_t>(sizeof(tempEFS.mBytesBuffer)));
+    EXPECT_EQ(memcmp(double_size_buffer, tempEFS.mBytesBuffer, sizeof(tempEFS.mBytesBuffer)), 0);
 
     tempEFS.Clear();
-    NL_TEST_ASSERT(aSuite, tempEFS.IsEmpty());
+    EXPECT_TRUE(tempEFS.IsEmpty());
 
     // Test insertion of uninitialized EFS
-    NL_TEST_ASSERT(aSuite, CHIP_ERROR_INVALID_ARGUMENT == EFS->InsertFieldSet(tempEFS));
-    NL_TEST_ASSERT(aSuite, 0 == EFS->GetFieldSetCount());
+    EXPECT_EQ(CHIP_ERROR_INVALID_ARGUMENT, EFS->InsertFieldSet(tempEFS));
+    EXPECT_EQ(0, EFS->GetFieldSetCount());
 
     // Test insertion of empty EFS
     tempEFS.mID = kOnOffClusterId;
-    NL_TEST_ASSERT(aSuite, CHIP_ERROR_INVALID_ARGUMENT == EFS->InsertFieldSet(tempEFS));
-    NL_TEST_ASSERT(aSuite, 0 == EFS->GetFieldSetCount());
+    EXPECT_EQ(CHIP_ERROR_INVALID_ARGUMENT, EFS->InsertFieldSet(tempEFS));
+    EXPECT_EQ(0, EFS->GetFieldSetCount());
 
     // test operators on multiple EFS struct
-    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == testEFS1.InsertFieldSet(EFS1));
-    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == testEFS1.InsertFieldSet(EFS2));
-    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == testEFS1.InsertFieldSet(EFS3));
+    EXPECT_EQ(CHIP_NO_ERROR, testEFS1.InsertFieldSet(EFS1));
+    EXPECT_EQ(CHIP_NO_ERROR, testEFS1.InsertFieldSet(EFS2));
+    EXPECT_EQ(CHIP_NO_ERROR, testEFS1.InsertFieldSet(EFS3));
 
-    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == testEFS2.InsertFieldSet(EFS3));
+    EXPECT_EQ(CHIP_NO_ERROR, testEFS2.InsertFieldSet(EFS3));
 
-    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == testEFS3.InsertFieldSet(EFS1));
-    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == testEFS3.InsertFieldSet(EFS2));
+    EXPECT_EQ(CHIP_NO_ERROR, testEFS3.InsertFieldSet(EFS1));
+    EXPECT_EQ(CHIP_NO_ERROR, testEFS3.InsertFieldSet(EFS2));
 
     tempTestEFS = testEFS1;
-    NL_TEST_ASSERT(aSuite, tempTestEFS == testEFS1);
-    NL_TEST_ASSERT(aSuite, !(tempTestEFS == testEFS2));
-    NL_TEST_ASSERT(aSuite, !(tempTestEFS == testEFS3));
+    EXPECT_EQ(tempTestEFS, testEFS1);
+    EXPECT_FALSE(tempTestEFS == testEFS2);
```

Comment:
Will EXPECT_NE work here?

Reply:
no it won't because the operator != is not overloaded for that class

Reply:
we have operator == but not != ... that is terrible!

Reply:
this is the case for a lot of the classes that we tested actually!

Reply:
should I add it for that class?